keras/src/optimizers/base_optimizer.py

```diff
@@ -6,11 +6,12 @@
 from keras.src import ops
 from keras.src.optimizers.schedules import learning_rate_schedule
 from keras.src.saving import serialization_lib
+from keras.src.saving.keras_pickleable import KerasPickleable
 from keras.src.utils import tracking
 from keras.src.utils.naming import auto_name
 
 
-class BaseOptimizer:
+class BaseOptimizer(KerasPickleable):
```

Comment:
The Pickleable thingy calls `_save_model_to_fileobj`. The name seems to indicate that it only works with models, but does it actually work for any serializable? If that's the case, then we should make the base layer class subclass Pickleable rather than Model.

Reply:
It seems like it works (unit tests pass? though perhaps I should do more rigorous testing like training a model with a round trip pickled optimizer?)

perhaps there are some edge cases I'm missing?  I can try adding to Layer as well, and see if there are any issues.

Reply:
It seems to work all over from what I can tell 👍 I think this is because under the hood they basically just call `serialize_keras_object`/`deserialize_keras_object`.